### PR TITLE
Shooting obstacle <feature

### PR DESCRIPTION
**Pull request by NilsHullegien**
Adds an obstacle that shoots 1 bullet (a zeppelin for now ;) ).

### CODE REVIEW

File: core/src/main/java/com/group5/core/world/Bullet.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                pos.x, pos.y,
+                0, 0,
+                getWidth(), getHeight(),
+                1, 1,
+                0,
+                0, 0,
+                (int) (getWidth() / WorldManager.PHYSICS_SCALE_FACTOR),
+                (int) (getHeight() / WorldManager.PHYSICS_SCALE_FACTOR),
+                true, false);
+        Body b = getPhysicsBody();
```
Comment (mwolting) [inline core/src/main/java/com/group5/core/world/Bullet.java]: This should really be in the update method

File: core/src/test/java/com/group5/core/ObstacleTest.java [hunk 2]
```diff
@@ -1,62 +1,66 @@
...
-     */
-    @Test
-    public void testEqualsSameProperties() {
-        Obstacle obstacle = new Obstacle(physicsWorld, new Vector2(0, 0));
-        Obstacle sameProp = new Obstacle(physicsWorld, new Vector2(0, 0));
-        assertTrue(obstacle.equals(sameProp));
-    }
-
-}
+//package com.group5.core;
```
Comment (mwolting) [inline core/src/test/java/com/group5/core/ObstacleTest.java]: If we're commenting all of this out, why not just remove it?

**Comment by NilsHullegien** [comment]
Suggestions were added in the latest commit.

**Comment by mwolting** [comment]
The bullets aren't really working reliably. The first few fire in time, but after a while they don't fire and sometimes pop up after the player has passed the enemy.

**Comment by mwolting** [comment]
Another suggestion: maybe throw a cap on the velocity of the bullet (if physicsbody.linearvelocity < max then applylinearimpulse) and give it a higher force, so that the speed is more constant and acceleration doesnt continue endlessly.

File: core/src/main/java/com/group5/core/levels/OnlyFloorLevel.java [hunk 3]
```diff
@@ -42,6 +43,8 @@ public OnlyFloorLevel(final GameSlice before, final World world) {
     private void createElems(final World world) {
         getElems().add(new FloorTile(world, new Vector2(this.getStartPoint().x / 50, 0)));
         float range = (this.getEndPoint().x - this.getStartPoint().x) / 50;
-        getElems().add(new Obstacle(world, new Vector2((float) (this.getStartPoint().x / 50 + range * Math.random()), 1.2f)));
+        float xPosObstacle = (float) (this.getStartPoint().x / 50 + range * Math.random());
+        getElems().add(new ShootingObstacle(world, new Vector2(xPosObstacle, 1.2f)));
```
Comment (levilime) [inline core/src/main/java/com/group5/core/levels/OnlyFloorLevel.java]: So the shootingobstacles should not be put in the GameSlice container but in a container for things that aren't attached to gameslices.

File: core/src/main/java/com/group5/core/levels/OnlyFloorLevel.java [hunk 4]
```diff
@@ -42,6 +43,8 @@ public OnlyFloorLevel(final GameSlice before, final World world) {
...
+        getElems().add(new Bullet(world, new Vector2(xPosObstacle, 2.f)));
```
Comment (levilime) [inline core/src/main/java/com/group5/core/levels/OnlyFloorLevel.java]: So the shootingobstacles should not be put in the GameSlice container but in a container for things that aren't attached to gameslices. When this container is implemented the robot shoot spawn the bullets in there)

**Comment by NilsHullegien** [comment]
Temporary close for fix merge conflicts.

**Pull request closed without merging**